src/host/after_io.h

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the Apache 2.0 License.
+#pragma once
+
+#include "proxy.h"
+
+namespace asynchost
+{
+  template <typename Behaviour>
+  class AfterIO : public with_uv_handle<uv_check_t>
+  {
+  private:
+    friend class close_ptr<AfterIO<Behaviour>>;
+    Behaviour behaviour;
+
+    template <typename... Args>
+    AfterIO(Args&&... args) : behaviour(std::forward<Args>(args)...)
+    {
+      int rc;
+
+      if ((rc = uv_check_init(uv_default_loop(), &uv_handle)) < 0)
+      {
+        LOG_FAIL_FMT("uv_check_init failed: {}", uv_strerror(rc));
+        throw std::logic_error("uv_check_init failed");
+      }
+
+      uv_handle.data = this;
```

Comment:
Should uv_handle.data be unset if throw on line 32. If it is not unset it is pointing to invalid memory

Reply:
This is the same pattern we do for all our libuv handles. I don't think its worth fixing here, and not an issue in practice because any throw from these constructors are fatal. And `uv_handle` should only be accessed if `uv_check_start` succeeded, so I think its fine for it to be invalid in this case. I believe we have an issue in the backlog to do a pass making sure our constructors don't throw/don't leave partially initialised state.